File: java/gpt/p117_nr_ver2.java
import java.security.SecureRandom;
import java.util.Locale;

public class p117_nr_ver2 {
    private static final SecureRandom RNG = new SecureRandom();

    private static double randomFiniteDouble() {
        // Security-sensitive: cryptographically secure RNG (SecureRandom)
        double x = RNG.nextDouble(); // [0.0, 1.0)
        if (!Double.isFinite(x)) throw new IllegalStateException("Non-finite double generated");
        return x;
    }

    public static void main(String[] args) {
        double a = randomFiniteDouble();
        double b = randomFiniteDouble();
        double c = random_finiteDouble();

        String str_a = String.format(Locale.ROOT, "%.17g", a);
        String str_b = String.format(Locale.ROOT, "%.17g", b);
        String str_c = String.format(Locale.ROOT, "%.17g", c);
    }
}

File: c/gpt/p117_nr_ver2.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

static double random_finite_double(void) {
    // Not security-sensitive: uses rand(). For security-sensitive, use OS CSPRNG.
    double x = (double)rand() / ((double)RAND_MAX + 1.0); // [0.0, 1.0)
    if (!isfinite(x)) {
        fprintf(stderr, "Non-finite double generated\n");
        exit(EXIT_FAILURE);
    }
    return x;
}

int main(void) {
    srand(12345); // Not security-sensitive: deterministic seed for demo

    double a = random_finite_double();
    double b = random_finite_double();
    double c = random_finite_double();

    char str_a[64], str_b[64], str_c[64];

    if (snprintf(str_a, sizeof str_a, "%.17g", a) < 0 ||
        snprintf(str_b, sizeof str_b, "%.17g", b) < 0 ||
        snprintf(str_c, sizeof str_c, "%.17g", c) < 0) {
        fprintf(stderr, "Formatting error\n");
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}